src/index.ts

```diff
@@ -570,6 +594,37 @@ export interface Redirects {
   [redirectQuery: string]: RedirectTarget;
 }
 
+/**
+ * PromotionPin defines a promotion pin.
+ */
+export interface PromotionPin {
+  key: {
+    field: string;
+    value: string;
+  };
+  position: number;
+}
```

Comment:
So, we can't rely on the position. James wisely pointed out that the pin position may not be the actual item position because they can collapse down when there are not enough organic items to fill the space that would otherwise appear between items which are spread apart.

Imagine an item pinned at `1` and and Item pinned at `12` on a category, which due to low product stock or whatever, now only has 5 other organic results. The actual result set would have a pinned item at 1, followed by 5 organic items, then the next pinned item at 7 instead of 12.

Also! We can't rely on `value` being equal to one of the result items `_id` field. We have to actually check the field names. The field and value will be _some_ field which is unique across the result set but not specifically the `_id` field. I'll make some notes above about the implications for this.

Reply:
But wait there's more! Not only can the field be something other than `_id`, it can also be different across pins.

Reply:
Right okay, I see. Yeah, the `_id` thing was something I wanted to ask about. I initially looked into the backend and considered that the key field might be any unique field and wrote the code accordingly, but then I looked into the console and saw that it was just using `_id` and changed it. 

Thanks for clearing that up.